algoliasearch-common/src/main/java/com/algolia/search/AsyncInsightsClient.java

```diff
@@ -0,0 +1,115 @@
+package com.algolia.search;
+
+import com.algolia.search.http.AlgoliaRequest;
+import com.algolia.search.http.AlgoliaRequestKind;
+import com.algolia.search.http.HttpMethod;
+import com.algolia.search.inputs.insights.InsightsEvent;
+import com.algolia.search.inputs.insights.InsightsRequest;
+import com.algolia.search.inputs.insights.InsightsResult;
+import com.algolia.search.objects.InsightsConfig;
+import com.algolia.search.objects.RequestOptions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+public class AsyncInsightsClient {
+
+  private AsyncAPIClient client;
+  private InsightsConfig config;
+  private String host;
+
+  /**
+   * @param applicationId The application ID
+   * @param apiKey The API Key
+   * @param client API Client
+   */
+  public AsyncInsightsClient(
+      @Nonnull String applicationId, @Nonnull String apiKey, @Nonnull AsyncAPIClient client) {
+    this(applicationId, apiKey, "us", client);
+  }
+
+  /**
+   * @param applicationId The applicationID
+   * @param apiKey The api KEY
+   * @param region The server region
+   * @param client APIClient
+   */
+  public AsyncInsightsClient(
+      @Nonnull String applicationId,
+      @Nonnull String apiKey,
+      @Nonnull String region,
+      @Nonnull AsyncAPIClient client) {
+    this(
+        new InsightsConfig().setApplicationId(applicationId).setApiKey(apiKey).setRegion(region),
+        client);
+  }
+
+  /**
+   * @param config InsightsClient
+   * @param client APIClient
+   */
+  public AsyncInsightsClient(@Nonnull InsightsConfig config, @Nonnull AsyncAPIClient client) {
+    this.config = config;
+    this.client = client;
+    this.host = "insights." + config.getRegion() + ".algolia.io";
```

Comment:
Is `getRegion` only used to set the Insights region or for Analytics as well? If it's only for Insights, could we rename the parameter and its getter `getInsightsRegion` to ensure we are talking about the Insights API. Also, if it's used for both Analytics and Insights API, I'm wondering if it makes sense to decorrelate them (probably not). What do you think @nunomaduro @julienbourdeau?